lambdas/tests/unit/handlers/access-token-handler.test.ts

```diff
@@ -1,270 +1,388 @@
+import { SSMClient } from "@aws-sdk/client-ssm";
+import { JwtVerifier, JwtVerifierFactory } from "../../../src/common/security/jwt-verifier";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { DynamoDBDocument } from "@aws-sdk/lib-dynamodb";
+import { SessionItem } from "../../../src/types/session-item";
+
+jest.mock("../../../src/common/config/config-service");
+jest.mock("../../../src/common/security/jwt-verifier");
+jest.mock("@aws-lambda-powertools/metrics");
+jest.mock("@aws-lambda-powertools/logger");
+jest.mock("@aws-sdk/lib-dynamodb", () => {
+    return {
+        __esModule: true,
+        ...jest.requireActual("@aws-sdk/lib-dynamodb"),
+        GetCommand: jest.fn(),
+        UpdateCommand: jest.fn(),
+    };
+});
+
+describe("access-token-handler.ts", () => {
+    const mockDynamoDbClient = jest.mocked(DynamoDBDocument);
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+        const impl = () => {
+            const mockPromise = new Promise<any>((resolve) => {
+                resolve({ Parameters: [] });
+            });
+            return jest.fn().mockImplementation(() => {
+                return mockPromise;
+            });
+        };
+        mockDynamoDbClient.prototype.send = impl();
+        mockDynamoDbClient.prototype.query = impl();
+    });
+
+    describe("Handler", () => {
+        let accessTokenLambda: AccessTokenLambda;
+        const configService = new ConfigService(jest.fn() as unknown as SSMClient);
+        const logger = new Logger();
+        const jwtVerificationConfig: JwtVerificationConfig = {
+            publicSigningJwk: "",
+            jwtSigningAlgorithm: "",
+        };
+        const jwtVerifier = new JwtVerifier(jwtVerificationConfig, logger);
+        const mockJwtVerifierFactory = jest.mocked(JwtVerifierFactory);
+        const mockConfigService = jest.mocked(ConfigService);
+        const accessTokenService = new AccessTokenService();
+        const sessionService = new SessionService(mockDynamoDbClient.prototype, configService);
+        const accessTokenRequestValidator = new AccessTokenRequestValidator(mockJwtVerifierFactory.prototype);
+
+        describe("success paths", () => {
+            beforeEach(() => {
+                jest.resetAllMocks();
+                configService.init = () => Promise.resolve();
+                accessTokenLambda = new AccessTokenLambda(
+                    accessTokenService,
+                    sessionService,
+                    accessTokenRequestValidator,
+                );
+                jest.spyOn(mockJwtVerifierFactory.prototype, "create").mockReturnValue(jwtVerifier);
+                jest.spyOn(jwtVerifier, "verify").mockReturnValue(
+                    new Promise<any>((resolve) => {
+                        resolve(true);
+                    }),
+                );
+            });
+
+            it("should pass when payload matches session - TEMP mocked verify function", async () => {
```

Comment:
Is this still a TEMP mocked function?

Reply:
Renamed the test as that message copied over from the old test.